crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,13 +1,175 @@
-use crate::{not_yet_implemented, FormatNodeRule, PyFormatter};
+use crate::comments;
+use crate::comments::{
+    dangling_node_comments, leading_comments, leading_node_comments, trailing_node_comments,
+};
+use crate::expression::parentheses::Parenthesize;
+use crate::prelude::*;
+use crate::statement::FormatRefWithRule;
+use crate::trivia::lines_before;
+use crate::{FormatNodeRule, PyFormatter};
 use ruff_formatter::{write, Buffer, FormatResult};
+use ruff_python_ast::node::AstNode;
+use ruff_python_ast::prelude::{ExceptHandler, ExceptHandlerExceptHandler, Ranged};
 use rustpython_parser::ast::StmtTry;
 
 #[derive(Default)]
 pub struct FormatStmtTry;
 
+#[derive(Copy, Clone, Default)]
+pub struct FormatExceptHandler;
+
+impl FormatRule<ExceptHandler, PyFormatContext<'_>> for FormatExceptHandler {
+    fn fmt(
+        &self,
+        item: &ExceptHandler,
+        f: &mut Formatter<PyFormatContext<'_>>,
+    ) -> FormatResult<()> {
+        let ExceptHandler::ExceptHandler(except_handler_except_handler) = item;
+        let ExceptHandlerExceptHandler {
+            range: _,
+            type_,
+            name,
+            body,
+        } = except_handler_except_handler;
+
+        let comments_info = f.context().comments().clone();
+        let leading_comments =
+            comments_info.leading_comments(except_handler_except_handler.as_any_node_ref());
+        if let [first_leading_comment, ..] = leading_comments {
+            if lines_before(
+                first_leading_comment.slice().start(),
+                f.context().contents(),
+            ) > 1
+            {
+                write!(f, [empty_line()])?;
+            }
+        }
+
+        leading_node_comments(except_handler_except_handler).fmt(f)?;
+
+        write!(f, [text("except")])?;
+
+        if let Some(type_) = type_ {
+            write!(
+                f,
+                [space(), type_.format().with_options(Parenthesize::IfBreaks)]
+            )?;
+            if let Some(name) = name {
+                write!(
+                    f,
+                    [
+                        space(),
+                        text("as"),
+                        space(),
+                        dynamic_text(name.as_str(), None)
```

Comment:
identifiers are now formattable on main, so you should be able to remove this workaround